src/stat.h

```diff
@@ -181,6 +182,7 @@ template <class DataType, class RecordType = DataType>
 class CStatHistory : public CStat<DataType, RecordType>
 {
 protected:
+    CCriticalSection cs;
     unsigned int op;
```

Comment:
can we use the more lightweight std::lock_guard rather than a recursive lock?

Reply:
you mean std::mutex also?  Sure... deadlock detection won't work, but that's ok here because no other locks are taken in the stat object.

Reply:
done